python/guide/packages/gluon/loss.md

```diff
@@ -79,38 +83,79 @@ def plot(x, y):
+
+L2Loss also called Meaned Squared Error is a regression loss function that computes the squared distances between the target values and the output of the neural network. It is defined as:
+
+$$ L = \frac{1}{2} \sum_i \vert {label}_i - {pred}_i \vert^2. $$
+
+Compared to L1, L2 loss it is a smooth function and it creates larger gradients for large loss values. However due to the squaring it puts high weight on outliers. 
+
 ```{.python .input}
 show_regression_loss(gloss.L2Loss())
 ```
 
-Then plot the classification losses with label values fixed to be 1.
+#### [Huber Loss](https://mxnet.incubator.apache.org/api/python/gluon/loss.html#mxnet.gluon.loss.HuberLosss)
+
+HuberLoss  combines advantages of L1 and L2 loss. It calculates a smoothed L1 loss that is equal to L1 if the adsolute error exceeds a threshold $$\rho$$, otherwise it is equal to L2. It is defined as:
+$$ 
+\begin{split}L = \sum_i \begin{cases} \frac{1}{2 {rho}} ({label}_i - {pred}_i)^2 &
+                   \text{ if } |{label}_i - {pred}_i| < {rho} \\
+                   |{label}_i - {pred}_i| - \frac{{rho}}{2} &
+                   \text{ otherwise }
+    \end{cases}\end{split}
+$$
 
 ```{.python .input}
-def show_classification_loss(loss):
-    x = nd.arange(-5, 5, .1)
-    y = loss(x, nd.ones_like(x))
-    plot(x, y) 
-    
-show_classification_loss(gloss.LogisticLoss())
+show_regression_loss(gloss.HuberLoss(rho=1))
 ```
 
+#### [Cross Entropy Loss with Sigmoid](https://mxnet.incubator.apache.org/api/python/gluon/loss.html#mxnet.gluon.loss.SigmoidBinaryCrossEntropyLoss)
+
+Binary Cross Entropy is a loss function used for binary classification problems e.g. classifying images into 2 classes. Cross entropy measures the difference between two propbaility distributions and it is defined as:
+$$\sum_i -{(y\log(p) + (1 - y)\log(1 - p))} $$
+Before the loss is computed a sigmoid activation is applied per default. You can avoid that by setting ```from_sigmoid``` to False. 
+
 ```{.python .input}
-show_classification_loss(gloss.HingeLoss() )
+show_classification_loss(gloss.SigmoidBinaryCrossEntropyLoss())
 ```
 
-## Cross Entropy Loss with Softmax
+#### [Cross Entropy Loss with Softmax](https://mxnet.incubator.apache.org/api/python/gluon/loss.html#mxnet.gluon.loss.SigmoidBinaryCrossEntropyLoss)
 
 In classification, we often apply the
 softmax operator to the predicted outputs to obtain prediction probabilities,
-and then apply the cross entropy loss against the true labels. Running these two
-steps one-by-one, however, may need to numerical instabilities. The `loss`
+and then apply the cross entropy loss against the true labels: 
+
+$$ \begin{align}\begin{aligned}p = \softmax({pred})\\L = -\sum_i \sum_j {label}_j \log p_{ij}\end{aligned}\end{align}
+$$
+
+Running these two
+steps one-by-one, however, may lead to numerical instabilities. The `loss`
 module provides a single operators with softmax and cross entropy fused to avoid
 such problem.
```

Comment:
Same warning as above: Worth a mention that you should avoid adding a sofmax at end of your network if using this loss with from_sigmoid=True, to avoid double softmax.

Reply:
done.